TASK: Add google-site-verification meta tag

Benjamin-K opened the pull request:
When using the google search console to have a better overview about search results of your site, till now you had to add the required meta tag (or verification file or header) yourself for each project. 

This pull request adds a property to the seo tags so you can easily add your google-site-verification meta tag. Output will be 

```html
<meta name="google-site-verification" content="yourGoogleSiteVerificationProperty" />
```

kdambekalns reviewed (commented):
Looks good… but… 😇

Having this as a node property seems wrong, as it's a site-wide thing. Maybe handle it in the settings instead?

Benjamin-K commented:
What's the problem with having it as a node property? I think this is much easier to maintain than the settings. Plus: Having it in the settings requires us to handle multi-site setups more carefully, which will not be the case with a node property.

Sebobo commented:
It only needs to be applied to the homepage right?
Then we could have it as a separate mixin and devs can set it as super type to their homepage NodeType.
And in Fusion we can add the condition that its only rendered if `node == site`.

Wdyt?

kdambekalns commented:
> What's the problem with having it as a node property? 

It can easily lead to confusion if added on more nodes than just the site node. What will Google think? What will users expect?

kdambekalns commented:
> It only needs to be applied to the homepage right? Then we could have it as a separate mixin and devs can set it as super type to their homepage NodeType.

That would be a nice addition!

I'd still love to be able to set it via settings. That way it can be populated from an env variable in different server environments.

> And in Fusion we can add the condition that its only rendered if `node == site`.

Awesome.

Benjamin-K commented:
Good catches. So i will create a new mixin for that and update the docs. The fusion does not have to be changed (except for the  `node == site` condition).

What would you expect when setting the property via settings? We should handle multi-site setups when having the property via settings, too.

How about:
```yaml
Neos:
  Seo:
    googleSiteVerification:
      'siteNodeIdentifier': 'googleSiteVerificationProperty'
```

Benjamin-K commented:
Another solution for the node property could be:

```yaml
'Neos.Seo:SeoMetaTagsMixin':
  properties:
    metaGoogleSiteVerification:
      ui:
        inspector:
          hidden: 'ClientEval: node.depth > 2'
```

Sebobo commented:
I personally don't want properties on nodes that I don't need. So I definitely prefer the mixin to control the feature individually.

kdambekalns commented:
> What would you expect when setting the property via settings?

It should be used as default value, I could then disable/hide the property and have it "settings-only". Or leave the property in place and it would override the setting.

> We should handle multi-site setups when having the property via settings, too.

Definitely, possibly for other settings as well. I like the way it's done here:

https://github.com/neos/neos-googleanalytics/blob/164bdc771862c44aa343138a155963c3aee5b83f/Configuration/Settings.yaml#L52-L62

Benjamin-K commented:
Ok, i updated my commit to have settings as fallback and a mixin that could be added to the nodeType of the site node. Also i updated the documentation. I did not yet implement the multi site setup via settings, but that should be ok for now i think.

Benjamin-K commented:
> Definitely, possibly for other settings as well. I like the way it's done here:
> 
> https://github.com/neos/neos-googleanalytics/blob/164bdc771862c44aa343138a155963c3aee5b83f/Configuration/Settings.yaml#L52-L62

I like that way, too. But that should be a separate PR i think. And the change should not be breaking if possible.

Benjamin-K commented on Resources/Private/Fusion/Metadata/GoogleSiteVerification.fusion:
```diff
@@ -0,0 +1,11 @@
+prototype(Neos.Seo:GoogleSiteVerification) < prototype(Neos.Fusion:Tag) {
+    tagName = 'meta'
+    @context.googleSiteVerification = ${q(node).property('metaGoogleSiteVerification') || Configuration.setting('Neos.Seo.google.siteVerification')}
```
Updated the prototype to extend from `Neos.Fusion:Component`.

Sebobo reviewed (approved):
Looks good to me by reading.

jonnitto commented on Resources/Private/Fusion/Metadata/GoogleSiteVerification.fusion:
```diff
@@ -0,0 +1,13 @@
+
+    renderer = afx`
+        <meta
+            @if.hasGoogleSiteVerification={props.googleSiteVerification}
```
Remove this line

jonnitto commented on Resources/Private/Fusion/Metadata/GoogleSiteVerification.fusion:
```diff
@@ -0,0 +1,13 @@
+prototype(Neos.Seo:GoogleSiteVerification) < prototype(Neos.Fusion:Component) {
+    @if.isSite = ${node == site}
```
`@if.isSiteAndHasGoogleSiteVerification = ${node == site && this.googleSiteVerification}`

Benjamin-K replied:
Used two `@if` statements here, but change is applied.
